server/src/domain/plugin_backend_handler.rs

```diff
@@ -0,0 +1,510 @@
+use std::collections::HashSet;
+
+use async_trait::async_trait;
+use chrono::NaiveDateTime;
+use ldap3_proto::{
+    proto::{LdapBindRequest, LdapExtendedRequest, LdapModifyRequest, LdapOp, LdapSearchRequest},
+    LdapSearchResultEntry,
+};
+use lldap_auth::{login, registration, types::UserId};
+use lldap_domain::{
+    requests::{
+        CreateAttributeRequest, CreateGroupRequest, CreateUserRequest, UpdateGroupRequest,
+        UpdateUserRequest,
+    },
+    schema::Schema,
+    types::{AttributeName, Group, GroupDetails, GroupId, LdapObjectClass, User, UserAndGroups},
+};
+use lldap_domain_handlers::handler::{
+    BackendHandler, BindRequest, GroupBackendHandler, GroupListerBackendHandler,
+    GroupRequestFilter, LoginHandler, ReadSchemaBackendHandler, RequestContext,
+    SchemaBackendHandler, UserBackendHandler, UserListerBackendHandler, UserRequestFilter,
+    WithContextHandler,
+};
+use lldap_domain_model::error::Result;
+use lldap_plugin_engine::api::{
+    arguments::{ldap_bind_result::BindResult, ldap_search_result::SearchResult},
+    handler::{PluginHandler, PluginHandlerEvents},
+    types::PluginContext,
+};
+use lldap_plugin_kv_store::store::PluginKeyValueStore;
+
+use crate::{
+    domain::{
+        ldap::utils::LdapInfo, opaque_handler::OpaqueHandler, plugin::backend::ServerBackendAPI,
+        sql_backend_handler::SqlBackendHandler,
+    },
+    infra::{ldap_handler::InternalSearchResults, tcp_backend_handler::TcpBackendHandler},
+};
+
+use tracing::instrument;
+
+#[derive(Clone, Debug)]
+pub struct PluginBackendHandler {
+    pub(crate) backend_handler: SqlBackendHandler,
+    pub(crate) backend_api: &'static ServerBackendAPI<SqlBackendHandler>,
```

Comment:
Why is it `&'static`? Why not keep the `Box`? Is it so that the `PluginContext` can keep a reference to it?
I know it's not a final review, but let's have a comment about the why, and let's be careful about what we put inside, since it doesn't get destroyed